elephant/spike_train_correlation.py

```diff
@@ -246,3 +249,408 @@ def __calculate_correlation_or_covariance(binned_sts, binary, corrcoef_norm):
+        cch_result = neo.AnalogSignalArray(
+            signal=counts.reshape(counts.size, 1),
+            units=pq.dimensionless,
+            t_start=(bin_ids[0] - 0.5) * st_1.binsize,
+            sampling_period=st_1.binsize)
+        # Return only the hist_bins bins and counts before and after the
+        # central one
+        return cch_result, bin_ids
+
+    def _cch_speed(st_1, st_2, win, mode, norm, border_corr, binary, kern):
+
+        # Check that the spike trains are binned with the same temporal
+        # resolution
+        if not st1.matrix_rows == 1:
+            raise AssertionError("Spike train must be one dimensional")
+        if not st2.matrix_rows == 1:
+            raise AssertionError("Spike train must be one dimensional")
+        if not st1.binsize == st2.binsize:
+            raise AssertionError("Bin sizes must be equal")
+
+        # Retrieve the array of the binne spik train
+        st1_arr = st1.to_array()[0, :]
+        st2_arr = st2.to_array()[0, :]
+        binsize = st1.binsize
+
+        # Convert the to binary version
+        if binary:
+            st1_arr = np.array(st1_arr > 0, dtype=int)
+            st2_arr = np.array(st2_arr > 0, dtype=int)
+        max_num_bins = max(len(st1_arr), len(st2_arr))
+
+        # Cross correlate the spiketrains
+
+        # Case explicit temporal window
+        if win is not None:
+            # Window parameter given in number of bins (integer)
+            if isinstance(win[0], int) and isinstance(win[1], int):
+                # Check the window parameter values
+                if win[0] >= win[1] or win[0] <= -max_num_bins \
+                        or win[1] >= max_num_bins:
+                    raise ValueError(
+                        "The window exceed the length of the spike trains")
+                # Assign left and right edges of the cch
+                l, r = win[0], win[1]
+            # Window parameter given in time units
+            else:
+                # Check the window parameter values
+                if win[0].rescale(binsize.units).magnitude % \
+                    binsize.magnitude != 0 or win[1].rescale(
+                        binsize.units).magnitude % binsize.magnitude != 0:
+                    raise ValueError(
+                        "The window has to be a multiple of the binsize")
+                if win[0] >= win[1] or win[0] <= -max_num_bins * binsize \
+                        or win[1] >= max_num_bins * binsize:
+                    raise ValueError("The window exceed the length of the"
+                                     " spike trains")
+                # Assign left and right edges of the cch
+                l, r = int(win[0].rescale(binsize.units) / binsize), int(
+                    win[1].rescale(binsize.units) / binsize)
+
+            # Cross correlate the spike trains
+            corr = np.correlate(st2_arr, st1_arr, mode='full')
+            counts = corr[len(st1_arr)+l+1:len(st1_arr)+1+r+1]
```

Comment:
It is not exactly in the spirit of PR #39. Here you calculate the cross-correlation between the whole spike trains, which for long spike trains can take considerable amount of time. Most of the time the correlation window of interest (`win`) is much shorter than the full length of both spike trains,  so that the `correlate` function returns arrays much longer than necessary.  In #39 we trim the spike trains such that the `convolve` (and so `correlate`) returns an array only with the bins of interest. In most cases this is a huge performance boost.